lib/Entities/NewsletterTemplateEntity.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace MailPoet\Entities;
+
+use MailPoet\Doctrine\EntityTraits\AutoincrementedIdTrait;
+use MailPoet\Doctrine\EntityTraits\CreatedAtTrait;
+use MailPoet\Doctrine\EntityTraits\UpdatedAtTrait;
+use MailPoetVendor\Doctrine\ORM\Mapping as ORM;
+use MailPoetVendor\Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity()
+ * @ORM\Table(name="newsletter_templates")
+ */
+class NewsletterTemplateEntity {
+  use AutoincrementedIdTrait;
+  use CreatedAtTrait;
+  use UpdatedAtTrait;
+
+  /**
+   * @ORM\ManyToOne(targetEntity="MailPoet\Entities\NewsletterEntity")
+   * @var NewsletterEntity|null
+   */
+  private $newsletter;
+
+  /**
+   * @ORM\Column(type="string")
+   * @Assert\NotBlank()
+   * @var string
+   */
+  private $name;
```

Comment:
I think the type of this property is not correct. Also, getter and setter have an incorrect type. `$name` can be null if I write:
```
  $t = new NewsletterTemplateEntity();
  $t->getName();
```

We should prevent creating the entity without the name or update the type.

This also applies to other properties.

Reply:
In DB this column is not nullable so I'd rather not make it nullable. What remains is adding it to a constructor which causes a bit of an issue in the incoming data processing - I either have to add an extra check for `$data['name']` and throw an exception or something. However, the same is done by the `Assert\NotBlank()` annotation so I solved it by using `$data['name'] ?? ''` to avoid duplicating the validation logic.